source/modulo_core/src/Cell.cpp

```diff
@@ -20,7 +22,11 @@ Cell::Cell(const rclcpp::NodeOptions& options) :
 }
 
 Cell::~Cell() {
-  this->on_shutdown();
+  RCUTILS_LOG_INFO_NAMED(get_name(), "Shutting down the node before destruction");
+  auto callback_code = this->on_shutdown(this->get_current_state());
+  if (callback_code == rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::FAILURE) {
+    RCUTILS_LOG_ERROR_NAMED(get_name(), "Error during the shutdown process, shutting down anyway.");
+  }
```

Comment:
`callback_code` value is not used for much here, plus there is both a `FAILURE` and `ERROR` case (though one is deprecated). Better would be:

```suggestion
  if (this->on_shutdown(this->get_current_state()) != rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS) {
    RCUTILS_LOG_ERROR_NAMED(get_name(), "Error during the shutdown process, shutting down anyway.");
  }
```